src/iologindata.cpp

```diff
@@ -832,6 +841,31 @@ bool IOLoginData::savePlayer(Player* player)
 		return false;
 	}
 
+	// save account storage values
+	if (g_game.getNumberOfPlayersByAccount(player->getAccount()) == 1) {
```

Comment:
same here, just check if account id key exist and save it

Reply:
What we want is to avoid saving the storage map several times, since we can save it only once, when in fact all the players of the same account have disconnected, otherwise every time a player disconnects we will be saving the storages, I suppose this is implemented like this in case several characters of the same account can connect at the same time

Reply:
> otherwise every time a player disconnects we will be saving the storages

Is this a problem? If the server crashes, at least *some* of the values will be updated from the session. It's a write-through cache of sorts...

Reply:
Okay. I have to agree with you. We can get rid of this function and just go like:
1. Load only once using `Game::getPlayerByAccount(uint32_t accountId)` [1].
2. Always store the values.

===
[1] Checking if a player is online is of no use.

Reply:
Okay. I like the write-through idea. However, we can't get rid of this function just yet. We need it in order to delete the account storage map object, once all players have logged out of the account.

To sum up, I believe the final solution would be as follows (I'll update the branch right away):
1. Load the values either using the new function or `Game::getPlayerByAccount(uint32_t acc) == nullptr`. And actually, I'll go with the second option. It is not exactly an O(n) operation when there are one or more players from the same account logged in into the game.
2. Always save the storages.
3. Delete the account storage map reference using the `Game::getNumPlayersByAccount(uint32_t acc)` method.